metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py

```diff
@@ -68,3 +73,21 @@ def get_key_prefix(s3_uri: str) -> str:
             f"Not an S3 URI. Must start with one of the following prefixes: {str(S3_PREFIXES)}"
         )
     return strip_s3_prefix(s3_uri).split("/", maxsplit=1)[1]
+
+
+def _group_s3_objects_by_dirname(
```

Comment:
we may set it "public"

```suggestion
def group_s3_objects_by_dirname(
```

Reply:
Opps,, thx for pointing it out!